src/test/unit/helperFunctions/test_yara_binary_search.py

```diff
@@ -3,30 +3,90 @@
 from helperFunctions import yara_binary_search, fileSystem
 from os import path
 
+from helperFunctions.config import get_config_for_testing
+from helperFunctions.yara_binary_search import YaraRuleError
+
+
+TEST_FILE_1 = 'binary_search_test'
+TEST_FILE_2 = 'binary_search_test_2'
+TEST_FILE_3 = 'binary_search_test_3'
+
+
+class MockCommonDbInterface:
+    def __init__(self, config):
+        self.config = config
+        self.config['data_storage']['firmware_file_storage_directory'] = path.join(fileSystem.get_test_data_dir(), TEST_FILE_1)
+
+    @staticmethod
+    def get_uids_of_all_included_files(_):
+        return [TEST_FILE_2, TEST_FILE_3]
+
+
+def mock_connect_to_enter(_, config=None):
+    if config is None:
+        config = {'data_storage': {}}
+    return yara_binary_search.YaraBinarySearchScannerDbInterface(config)
+
 
 class TestHelperFunctionsYaraBinarySearch(unittest.TestCase):
 
     def setUp(self):
+        yara_binary_search.YaraBinarySearchScannerDbInterface.__bases__ = (MockCommonDbInterface,)
+        yara_binary_search.ConnectTo.__enter__ = mock_connect_to_enter
+        yara_binary_search.ConnectTo.__exit__ = lambda _, __, ___, ____: None
         self.yara_rule = b'rule test_rule {strings: $a = "test1234" condition: $a}'
-        test_path = path.join(fileSystem.get_test_data_dir(), "binary_search_test")
+        test_path = path.join(fileSystem.get_test_data_dir(), TEST_FILE_1)
         test_config = {'data_storage': {'firmware_file_storage_directory': test_path}}
         self.yara_binary_scanner = yara_binary_search.YaraBinarySearchScanner(test_config)
 
     def test_get_binary_search_result(self):
-        result = self.yara_binary_scanner.get_binary_search_result(self.yara_rule)
-        self.assertEqual(result, {'test_rule': ['binary_search_test']})
+        result = self.yara_binary_scanner.get_binary_search_result((self.yara_rule, None))
+        self.assertEqual(result, {'test_rule': [TEST_FILE_1]})
+
+    def test_get_binary_search_result_error(self):
+        result = self.yara_binary_scanner.get_binary_search_result((b'invalid yara rule {}', None))
+        assert isinstance(result, YaraRuleError)
+
+    def test_get_binary_search_result_for_single_firmware(self):
+        yara_rule = b'rule test_rule_2 {strings: $a = "TEST_STRING!" condition: $a}'
+        result = self.yara_binary_scanner.get_binary_search_result((yara_rule, 'single_firmware'))
```

Comment:
You could test the propagation of the uid by changing the mocked get_uids_of_all_included_files method.